File: order-service/src/main/java/com/paulopacifico/orderservice/order/api/OrderConfirmedEvent.java
package com.paulopacifico.orderservice.order.api;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public record OrderConfirmedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        BigDecimal price,
        Integer quantity,
        OffsetDateTime occurredAt
) {
}

File: payment-service/src/main/kotlin/com/paulopacifico/paymentservice/messaging/api/OrderConfirmedEvent.kt
package com.paulopacifico.paymentservice.messaging.api

import java.math.BigDecimal
import java.time.OffsetDateTime
import java.util.UUID

data class OrderConfirmedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val price: BigDecimal,
    val quantity: Int,
    val occurredAt: OffsetDateTime,
)
